src/ngx_http_lua_args.c

```diff
@@ -19,6 +19,38 @@ static int ngx_http_lua_ngx_req_set_uri_args(lua_State *L);
 static int ngx_http_lua_ngx_req_get_post_args(lua_State *L);
 
 
+static ngx_inline int
+ngx_http_lua_valid_args(const u_char *src, size_t size)
+{
+    static uint32_t   map[] = {
+        0x00002401, /* 0000 0000 0000 0000  0010 0100 0000 0001 */
+
+                    /* ?>=< ;:98 7654 3210  /.-, +*)( '&%$ #"!  */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+
+                    /* _^]\ [ZYX WVUT SRQP  ONML KJIH GFED CBA@ */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+
+                    /*  ~}| {zyx wvut srqp  onml kjih gfed cba` */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+        0x00000000, /* 0000 0000 0000 0000  0000 0000 0000 0000 */
+    };
+
+    while (size) {
+        if (map[*src >> 5] & (1U << (*src & 0x1f))) {
+            return 0;
+        }
+
+        src++;
+        size--;
+    }
+    return 1;
+}
+
```

Comment:
Forgot to run `ngx-releng` to check this?

Reply:
It's still a single blank line? Seems like `ngx-releng` misses this issue for some reasons I don't know. We always need 2 blank lines to separate C function definitions. See https://openresty.org/en/c-coding-style-guide.html